### ISSUE TITLE
How to define own discover server and how to point the storageos bin towards this on premise service?

### ISSUE THREAD

**Issue by gioppoluca** [storageos/go-cli#129]
The need  is to be able to install a discover service on premise and to point the storageos binary to this installation.
Where is the config file and the variable to set to change the actual value used now that id the  https://discovery.storageos.cloud/clusters
Thanks
Luca

**Comment by Arau** [comment]
Hi @gioppoluca,

you can't define your own discover server. If your cluster can't access to the internet to get a token, I'd recommend to use a different discovery method. 

As describe in the [docs](https://docs.storageos.com/docs/install/prerequisites/clusterdiscovery) you can use the ip addresses of your nodes to set the JOIN. StorageOS runs with hostNetwork=true.  
It is not necessary to set all the nodes in the JOIN variable. Once a new StorageOS node can connect to a member of the cluster, the gossip protocol discovers the whole list of members. For high availability, it is recommended to set up as many as possible, so if one node is unavailable at the bootstrap process the next in the list will be queried.

**Comment by Arau** [comment]
Hi @gioppoluca,

I'd like to add more information about this. You currently can't define your own discovery server because the StorageOS container doesn't have the option to specify a non default discovery url. This is being fixed on the next release so an environmental variable will override the url.

**Comment by gioppoluca** [comment]
Since I'm just testing the stuff and at the moment I have only 1 node do I have to place just the IP of the node itsef?
Thanks
Luca

**Comment by Arau** [comment]
Yes, with one node you have to set the JOIN as itself, such as JOIN=$ADVERTISE_IP
